tensorflow/core/lib/io/buffered_random_access_file.h

```diff
@@ -0,0 +1,57 @@
+#ifndef TENSORFLOW_CORE_LIB_IO_BUFFERED_RANDOM_ACCESS_FILE_H_
+#define TENSORFLOW_CORE_LIB_IO_BUFFERED_RANDOM_ACCESS_FILE_H_
+
+#include <memory>
+
+#include "tensorflow/core/platform/file_system.h"
+
+#include "tensorflow/core/platform/mutex.h"
+namespace tensorflow {
+namespace io {
+
+/// \brief A RandomAccessFile with an internal buffer.
+class BufferedRandomAccessFile : public RandomAccessFile {
+ public:
+  /// \brief Create a BufferedRandomAccessFile with given buf_size and an owned
+  /// RandomAccessFile.
+  BufferedRandomAccessFile(std::unique_ptr<RandomAccessFile>&& other_file,
+                           size_t buf_size);
+
+  /// \brief Create a BufferedRandomAccessFile with given buf_size and a
+  /// RandomAccessFile which may be owned or not.
+  BufferedRandomAccessFile(RandomAccessFile* other_file, size_t buf_size,
```

Comment:
From the point of view of the user of the class, it is better to decide on a single ownership story. Either have this class own the `other_file` or not, but please don't do both as that opens up a lot of avenues for bugs and security nightmares.

Reply:
Totally agreed on the single ownership story. The second constructor was used for compatibility reasons for internal use. But I think it's OK to remove it in tensorflow core.